lib/awful/tooltip.lua

```diff
@@ -405,6 +412,21 @@ function tooltip:set_preferred_positions(value)
     set_geometry(self)
 end
 
+--- Set a widget to display.
+-- @tparam widget widget The widget to display
+-- @tparam integer width The width of the tooltip
+-- @tparam integer height The height of the tooltip
+function tooltip:set_widget(widget, width, height)
```

Comment:
> Also, the following widgets/layouts cannot be used in this scenario since they just take all 
> available space:

They would work with a constraint widget and/or forced_height/width. This is exactly how they work in a `wibar`. For the sake of consistency, I would still argue my previous comment stands.